tests/server/test_utils.py

```diff
@@ -0,0 +1,172 @@
+                sample_id="07452-01",
+                batch_id="201860",
+                ratio_13=0.92,
+                fetal_fraction=11.5,
+            ),
+            sample(
+                sample_id="07452-02",
+                batch_id="201860",
+                ratio_13=0.93,
+                fetal_fraction=12.5,
+            ),
+            sample(
+                sample_id="07452-03",
+                batch_id="201861",
+                ratio_13=0.94,
+                fetal_fraction=13.5,
+            ),
+            sample(
+                sample_id="07452-04",
+                batch_id="201861",
+                ratio_13=0.95,
+                fetal_fraction=14.5,
+            ),
+            sample(
+                sample_id="07452-05",
+                batch_id="201861",
+                ratio_13=0.96,
+                fetal_fraction=15.5,
+            ),
+            sample(
+                sample_id="07452-06",
+                batch_id="201862",
+                ratio_13=0.97,
+                fetal_fraction=16.5,
+            ),
+            sample(
+                sample_id="07452-07",
+                batch_id="201862",
+                ratio_13=0.98,
+                fetal_fraction=17.5,
+            ),
+            sample(
+                sample_id="07452-08",
+                batch_id="201862",
+                ratio_13=0.99,
+                fetal_fraction=18.5,
+            ),
+        ]
+    )
+
+    # WHEN running get_statistics_for_box_plot
+    results = get_statistics_for_box_plot(
+        adapter=app.adapter,
+        batches=["201860", "201861", "201862"],
+        fields=["Ratio_13", "FF_Formatted"],
+    )
+
+    # THEN the results shoule be:
+    expected_result = [
+        {
+            "FF_Formatted": [10.5, 11.5, 12.5],
+            "Ratio_13": [0.91, 0.92, 0.93],
+            "_id": {"batch": "201860", "date": "2022-03-10"},
+        },
+        {
+            "FF_Formatted": [13.5, 14.5, 15.5],
+            "Ratio_13": [0.94, 0.95, 0.96],
+            "_id": {"batch": "201861", "date": "2022-03-10"},
+        },
+        {
+            "FF_Formatted": [16.5, 17.5, 18.5],
+            "Ratio_13": [0.97, 0.98, 0.99],
+            "_id": {"batch": "201862", "date": "2022-03-10"},
+        },
+    ]
+    assert results == expected_result
+
+
+def test_get_statistics_for_scatter_plot(database):
+    app.db = database
+    app.adapter = NiptAdapter(database.client, db_name=database.name)
+
+    # GIVEN a list with three batch documents:
+    batches = [
+        batch(batch_id="201860", stdev_13=0.02, stdev_21=0.01),
+        batch(batch_id="201861", stdev_13=0.01, stdev_21=0.09),
+        batch(batch_id="201862", stdev_13=0.05, stdev_21=0.08),
+    ]
+
+    # WHEN running get_statistics_for_scatter_plot
+    results = get_statistics_for_scatter_plot(
+        batches=batches, fields=["Stdev_13", "Stdev_21"]
+    )
+
+    # THEN the results should look like this:
+    expected_result = {
```

Comment:
And check the results like this:

for res in expected_result.items():
 assert bla bla in some list you used to create the batches above

Reply:
👍